src/Storages/MutationCommands.cpp

```diff
@@ -182,4 +187,45 @@ void MutationCommands::readText(ReadBuffer & in)
     }
 }
 
+void MutationCommands::writeSettings(WriteBuffer & out) const
+{
+    for (auto it = settings_changes.begin(); it != settings_changes.end(); ++it)
+    {
+        if (it != settings_changes.begin())
+            out << ", ";
+        formatSettingName(it->name, out);
+        out << " = ";
+        writeEscapedString(applyVisitor(FieldVisitorToString(), it->value), out);
+    }
+}
+
+void MutationCommands::readSettings(ReadBuffer & in)
+{
+    String settings_str;
+    readEscapedString(settings_str, in);
+    ParserSetQuery p_settings(true);
+    ASTPtr ast = parseQuery(
+        p_settings, settings_str.data(), settings_str.data() + settings_str.length(), "mutation settings", 0, DBMS_DEFAULT_MAX_PARSER_DEPTH);
+    settings_changes = std::move(ast->as<ASTSetQuery>()->changes);
+}
+
+void MutationCommands::readSettingsFromQuery(ContextPtr context, const ASTPtr & settings_ast)
+{
+    static std::map<String, Field> settings_to_persist_in_mutation_entries({
+        {"update_allow_materialized_columns", Field(DEFAULT_UPDATE_ALLOW_MATERIALIZED_COLUMNS)},
```

Comment:
It should be possible to get default value from `Settings` (at least we have `resetValueToDefault`)